src/languages/java.md

```diff
@@ -1,3 +1,64 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is a general-purpose programming language, and one of the most popular in the world today. Platform.sh supports Java runtimes that can be used with build management tools such as Gradle, Maven, and Ant.
+
+
+## Supported versions
+
+### OpenJDK versions:
+
+* 8
+* 11
+* 12
+
+
+To select a Java version, specify a `type` such as `java:11`:
+
+```yaml
+# .platform.app.yaml
+type: "java:11"
+```
+
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) which handles decoding of service credential information for you.
+
+## Support build automation
+
+Platform.sh has support for software project management and comprehension tool in the Java World: 
```

Comment:
"Platform.sh supports the most common project management tools in the Java ecosystem, including:"

Reply:
Done